src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V62/Eth62ProtocolHandler.cs

```diff
@@ -154,13 +170,38 @@ GetBlockHeadersMessage getBlockHeadersMessage
                     HandleBodies(bodiesMsg, size);
                     break;
                 case Eth62MessageCode.NewBlock:
-                    NewBlockMessage newBlockMsg = Deserialize<NewBlockMessage>(message.Content);
-                    ReportIn(newBlockMsg);
-                    Handle(newBlockMsg);
+                    Metrics.Eth62NewBlockReceived++;
+                    if (IsPoS())
+                    {
+                        Disconnect(DisconnectReason.BreachOfProtocol, "NewBlock message received while PoS protocol activated.");
+                    }
+                    else
+                    {
+                        NewBlockMessage newBlockMsg = Deserialize<NewBlockMessage>(message.Content);
+                        ReportIn(newBlockMsg);
+                        Handle(newBlockMsg);
+                    }
                     break;
             }
         }
 
+        private bool IsPoS()
+        {
+            if (WasEverInPoS)
+            {
+                return true;
+            }
+            
+            if (_poSSwitcher.WasEverInPoS())
+            {
```

Comment:
is it ok to enter from two threads here?

Reply:
Shouldn't be a problem, it will then set WasEverInPoS to true twice and will unsubscribe from events in SyncServer twice (so in second try will just do nothing)